libs/client/ui/pages/src/lib/leaderboard/leaderboard.tsx

```diff
@@ -1,14 +1,147 @@
 import { CoinDepot, Header } from '@bregenz-bewegt/client-ui-components';
-import { IonContent, IonPage } from '@ionic/react';
+import {
+  leaderboardStore,
+  LeaderboardStore,
+  userStore,
+  UserStore,
+} from '@bregenz-bewegt/client/common/stores';
+import { Competitor } from '@bregenz-bewegt/shared/types';
+import {
+  IonCol,
+  IonContent,
+  IonGrid,
+  IonInfiniteScroll,
+  IonInfiniteScrollContent,
+  IonPage,
+  IonRow,
+  IonSelect,
+  IonSelectOption,
+  IonText,
+} from '@ionic/react';
+import { inject, observer } from 'mobx-react';
+import { useEffect, useState } from 'react';
 import './leaderboard.scss';
 
-export const Leaderboard: React.FC = () => {
-  return (
-    <IonPage>
-      <Header />
-      <IonContent className="leaderboard" fullscreen>
-        <CoinDepot />
-      </IonContent>
-    </IonPage>
-  );
-};
+const COMPETIORS_RELOAD_CHUNK_SIZE = 10;
+const MAX_SHOWN_COMPETITORS = COMPETIORS_RELOAD_CHUNK_SIZE * 10;
+
+export interface LeaderboardProps {
+  leaderboardStore?: LeaderboardStore;
+  userStore?: UserStore;
+}
+
+export const Leaderboard: React.FC<LeaderboardProps> = inject(
+  leaderboardStore.storeKey,
+  userStore.storeKey
+)(
+  observer(({ leaderboardStore, userStore }) => {
+    const [timespan, setTimespan] = useState<string>();
+    const [leaderboard, setLeaderboard] = useState<Competitor[]>([]);
+
+    useEffect(() => {
+      leaderboardStore
+        ?.fetch({ skip: 0, take: COMPETIORS_RELOAD_CHUNK_SIZE })
+        .then((data) => setLeaderboard(data))
+        .catch(() => {
+          setLeaderboard([]);
+        });
+    }, []);
+
+    const loadInfinite = (e: any) => {
+      if (leaderboard.length === MAX_SHOWN_COMPETITORS)
+        return e.target.complete();
+
+      leaderboardStore
+        ?.fetch({
+          skip: leaderboard.length,
+          take:
+            leaderboard.length + COMPETIORS_RELOAD_CHUNK_SIZE >
+            MAX_SHOWN_COMPETITORS
+              ? MAX_SHOWN_COMPETITORS - leaderboard.length
+              : COMPETIORS_RELOAD_CHUNK_SIZE,
+        })
+        .then((data) => {
+          setLeaderboard((prev) => orderLeaderboardDesc([...prev, ...data]));
+          e.target.complete();
+        })
+        .catch(() => {
+          setLeaderboard([]);
+          e.target.complete();
+        });
+    };
+
+    const orderLeaderboardDesc = (list: Competitor[]) => {
```

Comment:
Bruchts des unbedingt wenns scho im backend gesorted wird?

Reply:
jo weil jedes mol nur 10 datensätze vom backend krigt und ned alle wo denna in da liste dargestellt sin. wemmas im frontend ned generalsortiert sin die daten nur in da jeweiligen 10er chunks sortiert.

Reply:
ajo sry, du tuasch im Backend zersch taken und den ordern - was aber o net viel sin macht - müsst es net zersch "where>order>skip>take" sie, weil sunsch neamma jo einfach bloß zufällig die ersten 10 in da liste und sortieren sie den

Reply:
Stimmt hosch recht. des isch ma gar ned ufgfalla.

    this.prismaService.user.findMany({
      skip,
      take,
      where: { role: { not: Role.GUEST } },
      select: { username: true, coins: true },
      orderBy: { coins: 'desc' },
    });
    
so mach i aktuell da query. i muss ma des noml aschaua wia i do die reihenfolge vo da operatoren beeinfluss.